ionoscloud/resource_user_test.go

```diff
@@ -125,6 +125,16 @@ func TestAccUserBasic(t *testing.T) {
 						"name": "group1",
 					})),
 			},
+			{
+				Config:             testAccCheckNewUserGroup,
```

Comment:
@digna-ionos it isn't clear for me what we are testing here:
- in the beginning, we use `testAccCheckNewUserGroup` that basically creates and user and adds it to a group, `group1`
- then we remove the user from the group using `testAccRemoveUserFromGroup`
- then we check the user resource -- more specific we check that the `group_ids` set contains exactly one element, but I am not sure about this -- shouldn't it be 0 since we previously removed the user from the group?
- same for the data source

In order to check the functionality for this fix, I suggest leaving the test without the resource and data source checks, something like:
```
	{
				Config:             testAccCheckNewUserGroup,
				ExpectNonEmptyPlan: true,
				Check: resource.ComposeTestCheckFunc(
				testAccRemoveUserFromGroup(constant.GroupResource+".group1", constant.UserResource+"."+constant.NewUserResource),
			},
```
The indentation and syntax may not be ok, but the idea is that we want to see that a diff is generated when running this step:
- we create the user and add it to a group using `testAccCheckNewUserGroup` configuration
- we then remove the user from the group using `testAccRemoveUserFromGroup`
- after each step, a `terraform plan` is applied, in this case, the `terraform plan` will generate a diff (in the configuration the user appears to belong to a group, but when checking the remote state, the user doesn't belong to a group), but this is expected and because of `ExpectNonEmptyPlan: true` the test will pass.

Reply:
@digna-ionos forgot to mention, to make sure that this is ok, run the suggested configuration without `ExpectNonEmptyPlan: true`, the test should fail since `terraform plan` will suggest a diff. Then run it using `ExpectNonEmptyPlan: true` and the test should pass.